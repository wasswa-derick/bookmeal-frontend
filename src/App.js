import React from "react";
import { Route, Switch } from "react-router-dom";
import PropTypes from "prop-types";
import NotFound from "./NotFound";
import "./assets";
import { NavBar } from "./components/common";
import {
  Home,
  Signup,
  Login,
  BusinessSignup,
  Order,
  CustomerOrders
} from "./components";
import { GuestRoute, AdminRoute, AuthenticatedRoute } from "./routes";

import {
  Meals,
  Menus,
  AdminOrdersHistory,
  Dashboard,
  EditMeal,
  NewMenu,
  EditMenu
} from "./components/admin";

const App = ({ location }) => (
  <div>
    <NavBar />
    <Switch>
      <Route path="/" exact component={Home} />
      <GuestRoute path="/signup" exact component={Signup} />
      <GuestRoute
        location={location}
        path="/business/signup"
        exact
        component={BusinessSignup}
      />
      <GuestRoute location={location} path="/login" exact component={Login} />
      <AdminRoute
        location={location}
        path="/admin/meals"
        exact
        component={Meals}
      />
      <AdminRoute
        location={location}
        path="/admin/meal/:id/edit"
        exact
        component={EditMeal}
      />

      <AdminRoute
        location={location}
        path="/admin/menus"
        exact
        component={Menus}
      />

      <AdminRoute
        location={location}
        path="/admin/menus/new"
        exact
        component={NewMenu}
      />

      <AdminRoute
        location={location}
        path="/admin/menus/:id/edit"
        exact
        component={EditMenu}
      />

      <AdminRoute
        location={location}
        path="/admin/orders"
        exact
        component={AdminOrdersHistory}
      />

      <AdminRoute
        location={location}
        path="/admin"
        exact
        component={Dashboard}
      />

      <AuthenticatedRoute
        location={location}
        path="/menu/:id/order"
        exact
        component={Order}
      />

      <AuthenticatedRoute
        location={location}
        path="/orders"
        exact
        component={CustomerOrders}
      />
      <Route component={NotFound} />
    </Switch>
  </div>
);

App.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }).isRequired
};

export default App;
