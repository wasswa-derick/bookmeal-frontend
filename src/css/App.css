@import url("Common.css");
body {
    display: block;
    margin: 0px;
}

.app-banner {
    text-align: center;
    background: #f6f6f8;
    margin-top: -15px;
}

.app-banner h1 {
    font: 40px;
    color: #ffffff;
    font-weight: bolder;
}

.app-banner p {
    color: #000;
}

.app-banner img {
    width: 400px;
}

.specific-day {
    margin: 0px 20px 0 20px;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
}

.specific-day input {
    width: 50%;
}

.menu-items {
    margin: 0px 20px 0 20px;
}

.meal-option {
    display: inline-block;
    border: 1px solid #eee;
    margin: 10px;
    width: 260px;
    height: 300px;
}

.img-container {
    display: block;
    text-align: center;
}

.meal-option img {
    padding: 5px;
    align-self: center;
    width: 250px;
}

.meal-info {
    text-align: center;
}

.meal-info,
p,
{
    margin-left: 11px;
}

.meal-info .title {
    font-weight: bold;
}

.meal-info .price {
    color: #00a1e0;
}

.checkbox {
    border: 2px solid #e8e9eb;
    height: 10px;
    width: 10px;
}

.checkout-container {
    background: #f6f6f8;
    margin-bottom: 80px;
}

.checkout-container img {
    height: 100px;
}

.checkout-items {
    display: grid;
    align-items: center;
    justify-content: center;
}

.footer .text-muted {
    color: #00a1e0 !important;
}

.checkout {
    background: #f6f6f8;
    display: grid;
    grid-template-columns: 140px 200px 1fr;
    height: 100px;
    margin: 10px;
}

.checkout-img {
    display: grid;
    justify-content: center;
}

.checkout-btn {
    display: grid;
    align-content: center;
    margin-right: 50px;
    justify-content: end;
}

.checkout img {
    margin-top: 10px;
    height: 80px;
}

.checkout-totals {
    margin-right: 40px;
    display: grid;
    justify-content: center;
}

.checkout-totals label {
    margin-top: 10px;
    color: #d3d3d3;
}

.checkout p {
    font-weight: bold;
}

.order-total {
    margin-top: 20px;
    margin-bottom: 20px;
}

.order-total .amt {
    font-weight: bolder;
}

@media screen and (max-width: 480px) {
    .app-banner img {
        width: 300px;
    }
    .menu-items {
        display: grid;
        justify-content: center;
        justify-items: center;
    }
    .meal-option {
        display: inline-block;
        border: 1px solid #eee;
        margin: 20px 20px 10px 20px;
        width: 300px;
        height: 300px;
    }
    .checkout {
        background: #f6f6f8;
        display: grid;
        grid-template-columns: auto;
        grid-template-rows: repeat(3, 1fr);
        margin: 10px;
    }
    .checkout img {
        margin-top: 10px;
    }
    .checkout-img,
    .checkout-btn,
    .checkout-img {
        grid-row: span 1;
    }
    .checkout-btn {
        margin-right: 90px;
        margin-bottom: 30px;
    }
    .specific-day {
        display: block;
    }
    .specific-day input {
        width: 90%;
    }
}

@media screen and (max-width: 600px) {
    .navbar a:not(:first-child) {
        display: none;
    }
    .navbar a.icon {
        float: right;
        display: block;
    }
    .navbar.responsive {
        position: relative;
    }
    .navbar.responsive a.icon {
        position: absolute;
        right: 0;
        top: 0;
    }
    .navbar.responsive a {
        float: none;
        display: block;
        text-align: left;
    }
    .checkout-btn {
        margin-right: 90px;
        margin-bottom: 30px;
    }
}

@media screen and (max-width: 700 px) {
    .navbar a:not(:first-child) {
        display: none;
    }
    .navbar a.icon {
        float: right;
        display: block;
    }
    .navbar.responsive {
        position: relative;
    }
    .navbar.responsive a.icon {
        position: absolute;
        right: 0;
        top: 0;
    }
    .navbar.responsive a {
        float: none;
        display: block;
        text-align: left;
    }
}

@media screen and (max-width: 800px) {
    .navbar a:not(:first-child) {
        display: none;
    }
    .navbar a.icon {
        float: right;
        display: block;
    }
    .navbar.responsive {
        position: relative;
    }
    .navbar.responsive a.icon {
        position: absolute;
        right: 0;
        top: 0;
    }
    .navbar.responsive a {
        float: none;
        display: block;
        text-align: left;
    }
    .checkout-btn {
        margin-right: 130px;
        margin-bottom: 30px;
    }
}