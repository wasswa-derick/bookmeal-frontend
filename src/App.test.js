import { shallow } from "enzyme";
import React from "react";
import App from "./App";

describe("App Component", () => {
  it("renders  div", () => {
    const location = {
      pathname: ""
    };
    const wrapper = shallow(<App location={location} />);
    expect(wrapper.find("div").length).toEqual(1);
  });

  // TODO: add test for rendering correctly
});
