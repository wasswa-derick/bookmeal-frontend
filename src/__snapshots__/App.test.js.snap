// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
<div>
  <Connect(NavBar) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Connect(GuestRoute)
      component={[Function]}
      exact={true}
      path="/signup"
    />
    <Connect(GuestRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/business/signup"
    />
    <Connect(GuestRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/login"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin/meals"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin/meal/:id/edit"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin/menus"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin/menus/new"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin/menus/:id/edit"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin/orders"
    />
    <Connect(AdminRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/admin"
    />
    <Connect(AuthenticatedRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/menu/:id/order"
    />
    <Connect(AuthenticatedRoute)
      component={[Function]}
      exact={true}
      location={
        Object {
          "pathname": "",
        }
      }
      path="/orders"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
